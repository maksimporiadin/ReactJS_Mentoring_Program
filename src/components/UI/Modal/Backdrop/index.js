import React from 'react';
import './style.css';

const Backdrop = ({ show, close }) => (
    show ? <div className="backdrop" onClick={close}></div> : null
);

export default Backdrop;