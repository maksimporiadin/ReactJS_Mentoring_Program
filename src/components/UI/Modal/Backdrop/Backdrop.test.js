import 'jsdom-global/register';
import React from 'react';
import Backdrop from './index';
import { shallow } from 'enzyme';

describe('Backdrop', () => {
    let wrapper;
    const defaultProps = {
        close() {},
    };

    beforeEach(()=>{
        jest.resetAllMocks();
    });

    it('is renders show: true', () => {
        const props = {
            show: true,
        };

        wrapper = shallow(<Backdrop {...defaultProps} {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('is renders show: false', () => {
        const props = {
            show: false,
        };

        wrapper = shallow(<Backdrop {...defaultProps} {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});