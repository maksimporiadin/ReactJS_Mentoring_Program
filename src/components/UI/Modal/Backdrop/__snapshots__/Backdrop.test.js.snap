// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop is renders show: false 1`] = `""`;

exports[`Backdrop is renders show: true 1`] = `
<div
  className="backdrop"
  onClick={[Function]}
/>
`;
